src/credentials/activation.ts

```diff
@@ -29,8 +31,17 @@ export async function loginWithMostRecentCredentials(
     toolkitSettings: SettingsConfiguration,
     loginManager: LoginManager
 ): Promise<void> {
-    const previousCredentialsId = toolkitSettings.readSetting(profileSettingKey, '')
+    let previousCredentialsId = toolkitSettings.readSetting<string>(profileSettingKey, '')
     if (previousCredentialsId) {
+        // Migrate from older Toolkits - If the last providerId isn't in the new CredentialProviderId format,
+        // treat it like a Shared Crdentials Provider.
+        if (previousCredentialsId.indexOf(CREDENTIALS_PROVIDER_ID_SEPARATOR) === -1) {
+            previousCredentialsId = makeCredentialsProviderId({
+                credentialType: SharedCredentialsProviderFactory.CREDENTIAL_TYPE,
```

Comment:
it's weird that CREDENTIAL_TYPE is on the factory? It seems like there would be a more appropriate place to put it

Reply:
Good catch - At some point I was moving it over to the SharedCredentialsProvider, and forgot to finish that move, so its currently in both places. Moving it to the Provider.

I deliberately didn't make it somewhere central, because this feels more in line with being "open to extension", where generally the type is only relevant to the provider(s)/factories implementing it.